onyx/api/v3.py

```diff
@@ -121,5 +122,42 @@ def click():
     return handle_ping("click", api_version="3")
 
 
+@links.route('/activity-stream', methods=['POST'])
+@env.statsd.timer('v3_links_activity_stream')
+def activity_stream():
+    """Log activity stream ping sent from Firefox on each session"""
+    ip_addr = None
+    ua = None
+    ping_type = 'activity_stream'
+
+    try:
+        client_payload_raw = request.get_data(cache=False)
+        client_payload = ujson.decode(client_payload_raw)
+
+        ip_addr = request.headers.get('X-Forwarded-For')
+        if ip_addr is None:
+            ip_addr = request.remote_addr
+
+        ua = request.headers.get('User-Agent')
+        client_payload["ua"] = ua
+        client_payload["ip"] = ip_addr
+    except Exception:
+        env.log_dict(name="client_error", action="{0}_malformed_payload".format(ping_type), message={
+            "ip": ip_addr,
+            "ua": ua,
+            "ver": 3,
+        })
+
+        env.statsd.incr("{0}_error".format(ping_type))
+        return Response('', content_type='application/json; charset=utf-8',
+                        status=400)
+
+    env.log_dict(name="application", action="activity_stream", message=client_payload)
```

Comment:
You mean `user_event`?

We intentionally want to piggyback the `application` logs here, as the `user_event` requires us to put a lot of tile specific metrics in the payload, whereas application doesn't need that. Also, Infernyx will take care of sorting out the "activity-stream" pings from `application` logs.